File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int id=nums[0];
        int id1=nums[id];
        while(id!=id1){
            id=nums[id];
            id1=nums[nums[id1]];
        }
        id1=id;
        id=0;
        while(id!=id1){
            id=nums[id];
            id1=nums[id1];
        }
        return id;
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.py
import heapq;

class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        tmp = []
        for v in nums:
            if len(tmp) < k:
                heapq.heappush(tmp, v)
            elif v>tmp[0]:
                heapq.heappop(tmp)
                heapq.heappush(tmp, v)
        return tmp[0]